pydatastructs/miscellaneous_data_structures/queue.py

```diff
@@ -140,41 +141,40 @@ def __new__(cls, items=None, dtype=NoneType):
             if len(items) != 0 and dtype is NoneType:
                 obj._dtype = type(items[0])
             for x in items:
-                if type(x) == obj._dtype:
-                    obj.queue.append(x)
-                else:
-                    raise TypeError("Expected %s but got %s"%(obj._dtype, type(x)))
+                obj.append(x)
         else:
             raise TypeError("Expected type: list/tuple")
-        obj.front = obj.queue.head
-        obj.rear = obj.queue.tail
-        obj.size = obj.queue.size
         return obj
 
     def append(self, x):
         if self._dtype is NoneType:
             self._dtype = type(x)
-        elif type(x) is not self._dtype:
+        elif not _check_type(x, self._dtype):
             raise TypeError("Expected %s but got %s"%(self._dtype, type(x)))
```

Comment:
Remove this filter, as `LinkedListQueue` should allow dynamic types.

Reply:
ok